aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1095,17 +1095,17 @@ ReduceConfig setReduceConfig(const TensorIterator& iter){
 
   const int warp_split_threshold =
       std::min<int>(block_height * 16, max_values_per_thread);
+  bool split_across_warps = config.values_per_thread() >= warp_split_threshold;
   const int num_mp =
       at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
-  bool force_splitting_output = false;
 #ifdef USE_ROCM
-  force_splitting_output = iter.ndim() == 2 &&
+  bool force_splitting_output = iter.ndim() == 2 &&
       reduction_on_fastest_striding_dimension &&
       config.values_per_thread() < 1024 && num_mp < 100;
+  if (!force_splitting_output && split_across_warps) {
+#else
+  if (split_across_warps) {
```

Comment:
Is linter okay with this?  There are two `{` here so it confuses my vim :).  Can we replace the two with a single `{` go on its own line after the #endif?

Reply:
I will try to rewrite this to not have two `{` it looks weird to me too. :)